### PR TITLE
Add/apitoendpoints

### PR DESCRIPTION
**Pull request by CressChe**
https://trello.com/c/MOjEnJmn/48-add-api-to-the-beginning-of-all-backend-routes 

- branched from addapidata to test endpoint still works with /api/

### CODE REVIEW

**Review by oskarwilliams — COMMENTED** [review]
All good but you need to add it to the calls in the frontend

**Review by oskarwilliams — CHANGES_REQUESTED** [review]
Forgot to click request changes

File: WhaleSpotting/Controllers/OidcConfigurationController.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public OidcConfigurationController(IClientRequestParametersProvider clientReques
 
         public IClientRequestParametersProvider ClientRequestParametersProvider { get; }
 
-        [HttpGet("_configuration/{clientId}")]
+        [HttpGet("/api/_configuration/{clientId}")]
```
Comment (oskarwilliams) [inline WhaleSpotting/Controllers/OidcConfigurationController.cs]: This one shouldn't be changed

File: WhaleSpotting/react-app/src/api/apiClient.ts [hunk 2]
```diff
@@ -3,22 +3,22 @@ import authService from "../components/api-authorization/AuthorizeService";
 
 export async function fetchPendingSightings(pageNumber: number): Promise<SightingApiModel[]> {
     const headers = await getHeaders();
-    return await fetch(`https://localhost:5001/sightings/pending?page=${pageNumber}&pageSize=10`, {
+    return await fetch(`https://localhost:5001/api/sightings/pending?page=${pageNumber}&pageSize=10`, {
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/api/apiClient.ts]: as you are changing the others, can you make this
```suggestion
    return await fetch(`api/sightings/pending?page=${pageNumber}&pageSize=10`, {
```

File: WhaleSpotting/react-app/src/components/ReportSighting.tsx [hunk 3]
```diff
@@ -32,7 +32,7 @@ export default function ReportSighting(): JSX.Element {
         };
 
         let isError = false;
-        const response = await fetch("https://localhost:5001/sightings/create", {
+        const response = await fetch("https://localhost:5001/api/sightings/create", {
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/ReportSighting.tsx]: Same here 
```suggestion
        const response = await fetch("api/sightings/create", {
```

File: WhaleSpotting/react-app/src/components/SightingMapInfo.tsx [hunk 4]
```diff
@@ -69,7 +69,7 @@ export default function SightingMapInfo({ chosen }: SightingMapInfoProps): JSX.E
     }
 
     async function fetchSpecies(): Promise<Array<string> | void> {
-        await fetch(`https://localhost:5001/sightings/LocalSpecies?longitude=${chosen!.lon}&latitude=${chosen!.lat}`)
+        await fetch(`https://localhost:5001/api/sightings/LocalSpecies?longitude=${chosen!.lon}&latitude=${chosen!.lat}`)
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/SightingMapInfo.tsx]: And here
```suggestion
        await fetch(`api/sightings/LocalSpecies?longitude=${chosen!.lon}&latitude=${chosen!.lat}`)
```

File: WhaleSpotting/react-app/src/components/api-authorization/ApiAuthorizationConstants.js [hunk 5]
```diff
@@ -23,7 +23,7 @@ const prefix = "/authentication";
 
 export const ApplicationPaths = {
     DefaultLoginRedirectPath: "/",
-    ApiAuthorizationClientConfigurationUrl: `_configuration/${ApplicationName}`,
+    ApiAuthorizationClientConfigurationUrl: `/api/_configuration/${ApplicationName}`,
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/api-authorization/ApiAuthorizationConstants.js]: Remove the /api from here pls

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: WhaleSpotting/react-app/src/api/apiClient.ts (hunk 2 above)
Comment (CressChe) [reply WhaleSpotting/react-app/src/api/apiClient.ts]: :)

File: WhaleSpotting/react-app/src/components/ReportSighting.tsx (hunk 3 above)
Comment (CressChe) [reply WhaleSpotting/react-app/src/components/ReportSighting.tsx]: :)

File: WhaleSpotting/react-app/src/components/SightingMapInfo.tsx (hunk 4 above)
Comment (CressChe) [reply WhaleSpotting/react-app/src/components/SightingMapInfo.tsx]: :)

File: WhaleSpotting/react-app/src/components/api-authorization/ApiAuthorizationConstants.js (hunk 5 above)
Comment (CressChe) [reply WhaleSpotting/react-app/src/components/api-authorization/ApiAuthorizationConstants.js]: :)

**Review by oskarwilliams — APPROVED** [review]

**Pull request merged**